_protected/app/system/core/assets/cron/96h/BannedCoreCron.php

```diff
@@ -0,0 +1,269 @@
+		 * If we don't have a valid array to put address into, we create it.
+		 */
+		if (! is_array ( $this->ipNew )) {
+			$this->ipNew = array ();
+		}
+		/**
+		 * Call the webClient with the url
+		 */
+		$inbound = $this->webClient->get ( $url );
+
+		/**
+		 * Check we get a valid response
+		 */
+		if ($inbound->getStatusCode () !== '200') {
+			return false;
+		}
+
+		/**
+		 * Get the body and detach into a stream
+		 */
+		$bannedIPs = $inbound->getBody ()->detach ();
+
+		/**
+		 * Process the received IP
+		 */
+		while ( $bannedIP = fgets ( $bannedIPs ) ) {
+			/**
+			 * Trim the ip from return carriage and new line then add to the current array
+			 */
+			$this->ipNew [] = rtrim ( $bannedIP, "\n\r" );
+		}
+		return true;
+	}
+
+	/**
+	 * Process existing banned IP file and only keep valide IP adress
+	 */
+	private function processExistingIP() {
+		/**
+		 * We fill a temporary array with current adress
+		 */
+		$aBans = file ( $this->bannedFileName );
+		$this->ipOld = array ();
+		foreach ( $aBans as $ban ) {
+			/**
+			 * Array containing return IP adress
+			 *
+			 * @var array $ips
+			 */
+			$ips = preg_grep ( $this->ipRegExp, $ban );
+			/**
+			 * check if $ip empty in case we processed a text line
+			 */
+			if (! empty ( $ips )) {
+				/**
+				 * Use a foreach loop in case we have more than one IP per line
+				 */
+				foreach ( $ips as $ip ) {
+					$this->ipOld [] = $ip;
+				}
+			}
+		}
+	}
+
+	/**
+	 * Read both IPs array, merge and extract only unique one
+	 */
+	private function processIP() {
+		$newIP = array ();
+		$newIP = array_unique ( array_merge ( $this->ipNew, $this->ipOld ), SORT_STRING );
+		$this->ipNew = $newIP;
+	}
+
+	/**
+	 * Return a valid IPv4 regular expression
+	 * Using strict reject octal form (leading zero)
+	 *
+	 * @param bool $strict
+	 * @return string
+	 */
+	static public function regexpIP($strict = false) {
+		if (strict) {
+			/**
+			 * Regular Expression representing a valid IPv4 class address
+			 * Rejecting octal form (leading zero)
+			 *
+			 * @var string $ipRegExp
+			 */
+			$ipRegExp = '/(25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])\.';
+			$ipRegExp .= '(25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])\.';
+			$ipRegExp .= '(25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])\.';
+			$ipRegExp .= '(25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])/';
+		} else {
+			/**
+			 *
+			 * Regular Expression representing a valid IPv4 class address
+			 * We accept leading 0 but they normally imply octal so we shouldn't !!!
+			 *
+			 * @var string $ipRegExp
+			 */
+			$ipRegExp = '/(25[0-5]|2[0-9][0-9]|[01]?[0-9][0-9]?)\.';
+			$ipRegExp .= '(25[0-5]|2[0-9][0-9]|[01]?[0-9][0-9]?)\.';
+			$ipRegExp .= '(25[0-5]|2[0-9][0-9]|[01]?[0-9][0-9]?)\.';
+			$ipRegExp .= '(25[0-5]|2[0-9][0-9]|[01]?[0-9][0-9]?)/';
+		}
+		return $ipRegExp;
+	}
+
+	/**
+	 * Write IPs to banned ip file
+	 *
+	 * @return boolean
+	 */
+	private function writeIP() {
+		$outfile = fopen ( $this->bannedFileName, 'w+' );
```

Comment:
For better readability, you could use `file_put_contents` wrapper instead of using `fopen`, `fwrite` and `fclose` :